FHEM/90_SIGNALduino_un.pm

```diff
@@ -303,15 +271,14 @@ SIGNALduino_un_Attr(@)
   return if($a[0] ne "set" || $a[2] ne "IODev");
   my $hash = $defs{$a[1]};
   my $iohash = $defs{$a[3]};
-  my $cde = $hash->{CODE};
-  delete($modules{SIGNALduino_un}{defptr}{$cde});
-  $modules{SIGNALduino_un}{defptr}{$iohash->{NAME} . "." . $cde} = $hash;
+  my $cde = $hash->{DEF};
+  
+  #delete($modules{SIGNALduino_un}{defptr}{$cde});
```

Comment:
wie kann eine Definition entfernt werden, wenn diese Zeilen auskommentiert wurden?

Reply:
> wie kann eine Definition entfernt werden, wenn diese Zeilen auskommentiert wurden?

Die auskommentierte Zeile ist in der Attr Sub. Wenn diese Drin bleibt, so wird der defpr immer gelöscht beim Aufruf. Da die Sub ständig beim RESTART von FHEM durchlaufen wird, so würde die Zeile ständig mit den defpr löschen beim Neustarten, was zur Folge hat, das Device will neu angelegt werden obwohl es schon da ist. Mehrfach getestete und lange gesuchht bis ich den Fehler fand ;-)

>Die Attr-Funktion dient der Prüfung von Attributen, welche über den attr-Befehl gesetzt werden können.

Da muss das löschen nicht herein.


Das was du meinst, "wie kann eine Definition entfernt werden"  ist in der Sub Undef.
> Die Undef-Funktion wird aufgerufen wenn ein Gerät mit delete gelöscht wird  ....

Reply:
Okay, habe ich irgendwie falsch gesehen. Passt dann so :)